robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -313,21 +313,20 @@ $(document).ready(() => {
         let newGoalTemplate = $('#add-goal-coordinates').html()
 
         $("#goal-modal-body-content").append(newGoalTemplate)
-        $("#goal-modal-body-content .goal").addClass('goal-' + count).removeClass('goal')
-        $(".goal-" + count).find('*').addClass('goal-' + count)
+        $("#goal-modal-body-content .goal").addClass('goal-' + goalCount).removeClass('goal')
+        $(".goal-" + goalCount).find('*').addClass('goal-' + goalCount)
```

Comment:
The goal class is being used as an identifier. This is not proper JQuery design.

I don't see why you should need the identity. 

This class are in the form of something-NUMBER, which is not how classes are used. If you want to apply a style to all the goals you should have a class .goal, but you shouldnt have .goal-1, .goal-2 even to represent the 1st or 2nd goal. JQuery can fetch the ith child automatically